digdag-plugin-utils/src/main/java/io/digdag/util/Durations.java

```diff
@@ -10,7 +10,7 @@
 public class Durations
 {
     private static final Pattern PATTERN =
-            Pattern.compile("\\s*((?<hours>\\d+)\\s*h)?\\s*((?<minutes>\\d+)\\s*m)?\\s*((?<seconds>\\d+)\\s*s)?\\s*",
+            Pattern.compile("\\s*(?:(?<days>\\d+)\\s*d)?\\s*(?:(?<hours>\\d+)\\s*h)?\\s*(?:(?<minutes>\\d+)\\s*m)?\\s*(?:(?<seconds>\\d+)\\s*s)?\\s*",
```

Comment:
Not all components are required.

I.e., both `1d 2h 3m 4s` and `1d 5s` are valid durations.